hoomd/hpmc/UpdaterShape.h

```diff
@@ -0,0 +1,500 @@
+    detail::UpdateOrder m_update_order;      // order of particle types to apply the updater to
+    unsigned int m_instance = 0;             //!< Unique ID for RNG seeding
+    static constexpr Scalar m_tol = 0.00001; // minimum move size required to not be ignored.
+    };
+
+template<class Shape>
+UpdaterShape<Shape>::UpdaterShape(std::shared_ptr<SystemDefinition> sysdef,
+                                  std::shared_ptr<Trigger> trigger,
+                                  std::shared_ptr<IntegratorHPMCMono<Shape>> mc,
+                                  std::shared_ptr<ShapeMoveBase<Shape>> move)
+    : Updater(sysdef, trigger), m_mc(mc), m_move_function(move), m_global_partition(0),
+      m_determinant_inertia_tensor(m_pdata->getNTypes(), m_exec_conf),
+      m_ntypes(m_pdata->getNTypes(), m_exec_conf), m_initialized(false)
+    {
+    m_step_size.resize(m_pdata->getNTypes(), 0);
+    m_count_accepted.resize(m_pdata->getNTypes(), 0);
+    m_count_total.resize(m_pdata->getNTypes(), 0);
+    m_box_accepted.resize(m_pdata->getNTypes(), 0);
+    m_box_total.resize(m_pdata->getNTypes(), 0);
+    initializeDeterminatsInertiaTensor();
+    countParticlesPerType();
+
+    // TODO: add a sanity check to makesure that MPI is setup correctly
+    if (m_multi_phase)
+        {
+#ifdef ENABLE_MPI
+        MPI_Comm_rank(MPI_COMM_WORLD, &m_global_partition);
+        assert(m_global_partition < 2);
+#endif
+        }
+    }
+
+template<class Shape> UpdaterShape<Shape>::~UpdaterShape()
+    {
+    m_exec_conf->msg->notice(5) << "Destroying UpdaterShape " << std::endl;
+    }
+
+/*! Perform Metropolis Monte Carlo shape deformations
+\param timestep Current time step of the simulation
+*/
+template<class Shape> void UpdaterShape<Shape>::update(uint64_t timestep)
+    {
+    m_exec_conf->msg->notice(4) << "UpdaterShape update: " << timestep
+                                << ", initialized: " << std::boolalpha << m_initialized << " @ "
+                                << std::hex << this << std::dec << std::endl;
+    if (!m_initialized)
+        initializeDeterminatsInertiaTensor();
+
+    // Shuffle the order of particles types for this step
+    m_update_order.resize(m_pdata->getNTypes());
+    m_update_order.shuffle(timestep, this->m_sysdef->getSeed(), m_exec_conf->getRank());
+
+    Scalar log_boltz;
+    for (unsigned int i_sweep = 0; i_sweep < m_nsweeps; i_sweep++)
+        {
+        m_exec_conf->msg->notice(6) << "UpdaterShape copying data" << std::endl;
+        GPUArray<Scalar> determinant_inertia_tensor_old(m_determinant_inertia_tensor);
+        m_move_function->prepare(timestep);
+        auto& mc_params = m_mc->getParams();
+
+        for (unsigned int cur_type = 0; cur_type < m_type_select; cur_type++)
+            {
+            // make a trial move for i
+            unsigned int typ_i = m_update_order[cur_type];
+            // Skip move if step size is smaller than allowed tolerance
+            if (m_step_size[typ_i] < m_tol)
```

Comment:
```suggestion
            if (m_step_size[typ_i] != 0.0)
```
In very dense simulations, users may need to make very small moves. To disable moves for a type, users can set a step size exactly equal to 0.

Reply:
This doesn't appear to be resolved.